content/docs/tasks/external_auth_opa.md

```diff
@@ -0,0 +1,140 @@
+---
+title: "External Authorization"
+description: "Configure external authorization through MeshConfig."
+type: docs
+---
+# External Authorization
+
+## Overview
+External authorization allows offloading authorization of incoming HTTP requests to a remote endpoint not necessarily running inside the proxy's context.
+Proxies supporting external authorization will, for every request, issue a permission check to the remote endpoint. Due to the potential high number of RPS a system can have, external authorization has a number of tuneable settings to allow more or less control at the expense of performance (send HTTP payload or headers only, timeout, default behaviour if authorization fails to reply, etc.).
+
+OSM allows configuring envoy's [External Authorization extension](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/ext_authz_filter) through OSM's MeshConfig.
+
+## Limitations
+Currently, authorization filtering can't be dinamically configured to be installed in subsets or groups of proxies defined by the user, and is instead globally applied for all services in the mesh when enabled.
+
+Similarly, the filtering direction is to be statically applied to  `inbound` and `ingress` connections within the mesh, affecting any and all HTTP request made towards any service or application in the mesh when enabled.
+
+
+## OSM with OPA plugin external authorization walkthrough
+The following section will document how to configure external authorization in conjuction with `opa-envoy-plugin`.
+
+We strongly suggest to go through [OPA envoy plugin's](https://github.com/open-policy-agent/opa-envoy-plugin) documentation first to further understand their configuration options and deployment models.
+
+The following example uses a single, remote (over the network) endpoint to validate all traffic. This configuration is not recommended for a production deployment.
+
+- First, start by deploying OSM's Demo. We will use this sample deployment to test external authorization capabilities. Please refer to [OSM's Automated Demo](https://github.com/openservicemesh/osm/tree/main/demo#how-to-run-the-osm-automated-demo) and follow the instructions.
+
+```
+# Assuming OSM repo is available
+cd <PATH_TO_OSM_REPO>
+demo/run-osm-demo.sh  # wait for all services to come up
```

Comment:
@eduser25 Since the website primarily relies on the manual demo and corresponding manifests, it might make more sense to show OPA integration with the manual demo as opposed to the automated demo.

Reply:
guess I can make peace by just pointing to the manual demo for the sake of not going through all the steps again?
The end result for each is not much different from both demos after all I guess

Reply:
I'll leave it as is for now and leave an open issue on me to move it properly to manual when able.

Reply:
@eduser25 please open an issue to track this.